libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -686,10 +719,9 @@ std::pair<bool, std::vector<std::shared_ptr<Vote>>> VoteManager::checkRewardVote
   return {false, {}};
 }
 
-std::vector<std::shared_ptr<Vote>> VoteManager::getProposeRewardVotes() {
-  std::shared_lock lock(reward_votes_info_mutex_);
-  const auto reward_votes = getTwoTPlusOneVotedBlockVotes(reward_votes_period_, reward_votes_round_,
-                                                          TwoTPlusOneVotedBlockType::CertVotedBlock);
+std::vector<std::shared_ptr<Vote>> VoteManager::getRewardVotes() {
+  auto reward_votes = getTwoTPlusOneVotedBlockVotes(reward_votes_period_, reward_votes_round_,
```

Comment:
no need for mutex protection ?

Reply:
getTwoTPlusOneVotedBlockVotes has a mutex protection inside

Reply:
Fixed

Reply:
you have to put the lock before calling getTwoTPlusOneVotedBlockVotes, not after... and then you will again lock the same lock inside getTwoTPlusOneVotedBlockVotes. Based on usage seems like the orig. lock that protected just this info regarding period, round and hash of reward votes was better approach as these are separate info unrelated to verified_votes and its lock

Reply:
Fixed with the single mutex approach now. The problem with having two mutexes as before is that now with my change there are a lot of places in the code where we would need to lock both together or one after another and I fear that increases chance for a deadlock.

Reply:
no need to worry about deadlock - this new mutex is locked only in single place when we reset reward votes so it is very easy to check for any possibility of deadlock... based on some of the changes in this PR, which were made just because of this single lock usage, it sort of shows the bad design

Reply:
one of the things I did not like in previous code was that we are locking both locks at the same time. This always introduces risk of deadlock. I have returned it back to using two mutexes but with some changes to never lock it together